File: src/main/scala/bitlap/sbt/analyzer/model/DependencyGraphObject.scala
package bitlap.sbt.analyzer.model

final case class DependencyGraphObject(
  _gvid: Long,
  name: String,
  height: String,
  label: String,
  pos: String,
  width: String
) {

  def toDependency: Dependency = name match
    case DependencyGraphObject.regex(group, artifact, version) => Dependency(_gvid, group, artifact, version)
    case _                                                     => null
}

object DependencyGraphObject {
  val regex = "(.*):(.*):(.*)".r

}

File: src/main/scala/bitlap/sbt/analyzer/model/DependencyGraphObject.java
package bitlap.sbt.analyzer.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * @author 梦境迷离
 * @version 1.0, 2023/8/3
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class DependencyGraphObject {
    Integer _gvid;
    String name;

    public DependencyGraphObject() {
    }

    public Integer get_gvid() {
        return _gvid;
    }

    public void set_gvid(Integer _gvid) {
        this._gvid = _gvid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

}
